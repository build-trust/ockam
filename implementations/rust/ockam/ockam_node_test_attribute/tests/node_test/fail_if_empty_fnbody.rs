// This test checks that #[ockam_node_test_attribute::node_test] causes a compile time error
// if the function is passed a parameter of type `ockam::Context` but is unused.

#[ockam_node_test_attribute::node_test]
async fn my_test(_ctx: ockam::Context) -> ockam::Result<()> {}
