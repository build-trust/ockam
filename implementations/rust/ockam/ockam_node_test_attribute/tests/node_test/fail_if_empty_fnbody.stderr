error: Function body Cannot be Empty.
 --> $DIR/fail_if_empty_fnbody.rs:5:10
  |
5 | async fn my_test(_ctx: ockam::Context) -> ockam::Result<()> {}
  |          ^^^^^^^

error[E0601]: `main` function not found in crate `$CRATE`
 --> $DIR/fail_if_empty_fnbody.rs:4:1
  |
4 | / #[ockam_node_test_attribute::node_test]
5 | | async fn my_test(_ctx: ockam::Context) -> ockam::Result<()> {}
  | |______________________________________________________________^ consider adding a `main` function to `$DIR/tests/node_test/fail_if_empty_fnbody.rs`
